### DISCUSSION TITLE
Slack Notification Channel: support alternative API URL for Slack

### DISCUSSION THREAD

**Discussion by solidspark** [grafana/grafana#44876] [Alerting]
👋 Hello Grafana!

Slack is developing a government-focused product (https://slack-gov.com). The API for our government Slack product née GovSlack is located at https://api.slack-gov.com, and not https://api.slack.com. Grafana uses this API to emit Slack notifications from alerts through notification channels, using either a bot token or a legacy webhook.

To expand on that, we discovered that the API endpoint in slack.go is hardcoded: 

https://github.com/grafana/grafana/blob/f6c27b637e8570dac4ee7cbf9b71982c92e29627/pkg/services/alerting/notifiers/slack.go#L124

Therefore, if you were a government customer with a Grafana installation, you would have no way of using Slack notification channels with a GovSlack bot token.

Proposal 1:
* Allow `slackAPIEndpoint` to be set by config in grafana.ini

Proposal 2:
* Add a new field that overrides the domain name in `slackAPIEndpoint` in each notification channel.

Proposal 3:
In the Token field, add a checkbox, "This is a GovSlack token" and add a new const, and select the proper URL based on the value of the checkbox
```
const govSlackAPIEndpoint = "https://slack-gov.com/api/chat.postMessage"
```

Thank you! This will allow our customers to continue to use the Grafana integration with GovSlack.

**Comment by JohnnyQQQQ** [comment]
👋  Hey @solidspark! We discussed this internally and would like to move forward with proposal 2 as it's a more generic approach to the problem.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
